pulp_rpm/app/tasks/synchronizing.py

```diff
@@ -372,6 +448,18 @@ def newpkgcb(pkgId, name, arch):
                                 deferred_download=self.deferred_download
                             )
                             dc = DeclarativeContent(content=package, d_artifacts=[da])
+
+                            # find if a package relates to a modulemd
+                            if dc.content.nevra in nevra_to_module.keys():
+                                if 'modulemd_relation' not in dc.extra_data:
+                                    dc.extra_data = defaultdict(list)
```

Comment:
It can be created right after the declarative version is created, to avoid some if checks here.

Reply:
moved at package declarative content creation, saves lot of other lines